content/post/gld.md

```diff
@@ -0,0 +1,276 @@
+$ unzip ../dataset.zip
+$ # src-d/go-license-detector
+$ time license-detector * \
+  | grep -Pzo '\n[-0-9a-zA-Z]+\n\tno license' | grep -Pa '\tno ' | wc -l
+$ # benbalter/licensee
+$ time ls -1 | xargs -n1 -P4 licensee \
+  | grep -E "^License: Other" | wc -l
+$ # google/licenseclassifier
+$ time find -type f -print | xargs -n1 -P4 identify_license \
+  | cut -d/ -f2 | sort | uniq | wc -l
+$ # boyter/lc
+$ time lc . \
+  | grep -vE 'NOASSERTION|----|Directory' | cut -d" " -f1 | sort | uniq | wc -l
+```
+
+## Algorithm
+
+We have implemented license detection based on the `LICENSE` and `README` files for now, and wish to
+add fine-grained scanning of source code files in the future (do you want to help us?
+work on the [issue](https://github.com/src-d/go-license-detector/issues/24)). Given the
+stated license text, we compare it to the texts in the SPDX database and record the match. The naive
+way of making the comparisons is to calculate the
+[Levenshtein distance](https://en.wikipedia.org/wiki/Levenshtein_distance)
+between the query and each of the examples. This is not fast at all for two reasons:
+
+1. The complexity is linear to the number of examples.
+2. The complexity is linear to the size of the difference. Since we've always gotten very few matches,
+the rest of the distances will be slow to calculate.
+
+These two reasons confidently render the naive approach unusable.
+
+The core of go-license-detector's detection mechanism is [Locality Sensitive Hashing](https://en.wikipedia.org/wiki/Locality-sensitive_hashing).
+We treat each license as a [TF-IDF-weighted](https://en.wikipedia.org/wiki/Tf%E2%80%93idf)
+[bag-of-words](https://en.wikipedia.org/wiki/Bag-of-words_model) - that is, as a set of words where every
+word has a weight corresponding to the frequency in the license text (term frequency) and
+throughout the whole database (document frequency). This is a proven approach for large scale
+similarity detection which we used multiple times in the past. Although 400 items is clearly
+not large scale at all, it still makes sense to employ LSH because of the O(1) complexity
+guarantee. We saw that it works reasonably well in practice and introduces a small overhead,
+around 20MB of memory for the hashes and the vocabulary.
+
+The hashing algorithm which we use is
+[Weighted MinHash](http://static.googleusercontent.com/media/research.google.com/en//pubs/archive/36928.pdf)
+- again, battle-tested in the past, e.g. in [Apollo](https://github.com/src-d/apollo) or
+[bags deduplication](https://blog.sourced.tech/post/minhashcuda/).
+After careful tuning of false positives, false negatives, and performance, we decided to set the Jaccard
+similarity threshold for our algorithm to 75%, and the hash length to 154 samples.
+Since we discard the text structure by treating sequences as sets, we further calculate the Levenshtein
+distance to the matched database records in order to determine the precise confidence value.
```

Comment:
I would add a sentence that explains that you're only calculating the Levenshtein distance on matches from weighted minhash.